zokrates_cli/src/ops/mpc/verify.rs

```diff
@@ -0,0 +1,104 @@
+use crate::constants::{FLATTENED_CODE_DEFAULT_PATH, MPC_DEFAULT_PATH};
+use clap::{App, Arg, ArgMatches, SubCommand};
+use phase2::parameters::MPCParameters;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+use zokrates_core::ir;
+use zokrates_core::ir::ProgEnum;
+use zokrates_core::proof_system::bellman::Computation;
+use zokrates_field::Bn128Field;
+
+pub fn subcommand() -> App<'static, 'static> {
+    SubCommand::with_name("verify")
+        .about("Verifies correctness of MPC parameters, given a circuit instance")
+        .display_order(4)
+        .arg(
+            Arg::with_name("input")
+                .short("i")
+                .long("input")
+                .help("Path of the MPC parameters")
+                .value_name("FILE")
+                .takes_value(true)
+                .required(false)
+                .default_value(MPC_DEFAULT_PATH),
+        )
+        .arg(
+            Arg::with_name("circuit")
+                .short("c")
+                .long("circuit")
+                .help("Path of the circuit binary")
+                .value_name("FILE")
+                .takes_value(true)
+                .required(false)
+                .default_value(FLATTENED_CODE_DEFAULT_PATH),
+        )
+        .arg(
+            Arg::with_name("radix-dir")
+                .short("r")
+                .long("radix-dir")
+                .help("Path of the directory containing parameters for various 2^m circuit depths (phase1radix2m{0..=m})")
+                .value_name("PATH")
+                .takes_value(true)
+                .required(true),
+        )
+}
+
+pub fn exec(sub_matches: &ArgMatches) -> Result<(), String> {
+    // read compiled program
+    let path = Path::new(sub_matches.value_of("circuit").unwrap());
+    let file =
+        File::open(&path).map_err(|why| format!("Could not open `{}`: {}", path.display(), why))?;
+
+    let mut reader = BufReader::new(file);
+
+    match ProgEnum::deserialize(&mut reader)? {
+        ProgEnum::Bn128Program(p) => cli_mpc_verify(p, sub_matches),
+        _ => unimplemented!(),
```

Comment:
This should be unreachable as we should check this is allowed before calling this, cf `helpers.rs`

Reply:
yeah nice catch, I forgot to fix this :)